hardware/arduino/sam/cores/arduino/wiring_digital.c

```diff
@@ -90,6 +114,14 @@ extern void digitalWrite( uint32_t ulPin, uint32_t ulVal )
 
 extern int digitalRead( uint32_t ulPin )
 {
+  if ((g_pinStatus[ulPin] & 0xF) == PIN_STATUS_DIGITAL_OUTPUT) {
+    return (g_pinStatus[ulPin] & 0xF0) >> 4;
+  }
+
+  if ((g_pinStatus[ulPin] & 0xF) == PIN_STATUS_ANALOG) {
+    pinMode(ulPin, INPUT);
```

Comment:
what is the meaning of calling pinMode here?
I may have missed something but I understood the PIN_STATUS_ANALOG means the pin is under control of ADC or DAC. So, calling pinMode() which active the clock on PIOx and putting the PIOx pin as INPUT will run useless code.